Data/Nat/Le/total.kind2

```diff
@@ -0,0 +1,9 @@
+Data.Nat.Le.total (n: Data.Nat) (m: Data.Nat): Data.Either (Data.Nat.Le n m) (Data.Nat.Le m n)
```

Comment:
that is a good idea! I saw your comment after implementing the function that derives the Decidable proof automatically, but I wasn't liking how it required the Trait.Eq trait and ran the Decidable equality test again(https://github.com/HigherOrderCO/Kindex/pull/111/files#diff-d1e614236d07d323e1649b25606ecfa6b4b1a002093a9cabb687b7fc6c7c3101R20).

after reading your comment I tried implementing a disjunction type:
```
// inclusive or
#derive[match]
type Math.Relation.Disjunction (l: Type) (r: Type) {
  left  (val: l)
  right (val: r)
  both  (left: l) (right: r)
}
```
and making the total function return that instead. however what i require to derive the proof automatically more succintly would be something like:
```
type NoPrettyNameForThisComplicatedStructure (t) (rel: Math.Relation t) (a:t) (b:t) {
  left (val: rel a b) (neq: Prop.Not (Prop.Equal a b))
  right (val: rel b a) (neq: Prop.Not (Prop.Equal a b))
  both (left: rel a b) (right: rel b a)
```

because i need proof of the inequality when this structure returns right(as seen in https://github.com/HigherOrderCO/Kindex/pull/111/files#diff-d1e614236d07d323e1649b25606ecfa6b4b1a002093a9cabb687b7fc6c7c3101R20)